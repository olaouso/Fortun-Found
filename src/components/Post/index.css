.cardBody {
  color: #026fc2;
  font-size: 1em;
}
.postName,
.postDate {
  color: #868686;
  margin: 0.1;
}
.singlePost {
  font-family: "Acme";
  margin: 0.5em;
  padding: 0.2em;
  border: 0.1em solid #eef2f4;
  border-radius: 0.5em;
  background-color: #eef2f4;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.cardImg {
  width: 15em;
  height: 15em;
}

.cardTitle {
  color: #590995;
  font-size: 1.2em;
  margin: 0.4em;
}

.singlePost:hover {
  top: -0.2em;
}
