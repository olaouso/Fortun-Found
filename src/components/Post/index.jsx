/* eslint-disable react/jsx-one-expression-per-line */
import React from "react";
import { useTranslation } from "react-i18next";
import { Col } from "react-bootstrap";
import { Link } from "react-router-dom";
import "./index.css";
import loadingGif from "../PostList/image/loadingGif.gif";

export default function Post(postInfo) {
  const { t } = useTranslation();
  if (postInfo.postInfo && postInfo.postInfo.date && postInfo.postInfo.title) {
    return (
      <Col className="singlePost" lg={3} sm={12} md={6}>
        <Link to={`posts/${postInfo.type}/${postInfo.postInfo.id}`}>
          <p className="cardTitle text-center">{postInfo.postInfo.title}</p>
        </Link>
        <p className="postDate">
          {t("postPage.date")}: {postInfo.postInfo.date}
        </p>
        <p className="postName">
          {t("postPage.Author")}: {postInfo.postInfo.profileName}
        </p>
        <p className="cardBody text-center">{postInfo.postInfo.description}</p>
        <img
          className="cardImg"
          src={postInfo.postInfo.image}
          alt={t("postPage.altVolunteerHeader")}
        />
      </Col>
    );
  }
  // eslint-disable-next-line no-else-return
  else {
    return <img className="loadingGif" src={loadingGif} alt="loading..." />;
  }
}
