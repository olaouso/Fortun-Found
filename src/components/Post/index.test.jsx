import React from "react";
import renderer from "react-test-renderer";
import Post from "./index";

it("renders single Post component correctly", () => {
  const tree = renderer.create(<Post />).toJSON();
  expect(tree).toMatchSnapshot();
});
