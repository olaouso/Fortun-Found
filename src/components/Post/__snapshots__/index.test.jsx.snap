// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders single Post component correctly 1`] = `
<img
  alt="loading..."
  className="loadingGif"
  src="loadingGif.gif"
/>
`;
